bindings/rascal/utils/filter.py

```diff
@@ -504,7 +504,6 @@ def __init__(
             }
 
         else:
-            assert act_on == "feature"
```

Comment:
I think we want to keep this. Even if the validity check is done in the super class we are still assuming this to be true in that class. This assert would catch case like us adding a new "act_on" mode and forgetting to update this specific class

Reply:
Yeah, but it was breaking the tests, so I took it out. I don't like having two separate checks that each look like copypasta of each other.

Reply:
I prefer writing assert as defensively as possible, but I can see how you could have different preferences here =). I think we should fix the test instead of changing the code, but I can leave with this as well.

Reply:
I agree with the concept of defensive programming in general; I just don't like to do it with asserts because the error messages are so unhelpful (compare this with the ValueError raised in the superclass that explicitly tells the user what they did wrong). I guess if we really wanted to do this properly, we'd implement a superclass method `check_mode(self, valid_modes)`, given a list of valid modes, and let each subclass use it to throw the error.